pkg/planner/core/operator/logicalop/logical_join.go

```diff
@@ -1870,9 +1871,18 @@ func deriveNotNullExpr(ctx base.PlanContext, expr expression.Expression, schema
 
 // Conds2TableDual builds a LogicalTableDual if cond is constant false or null.
 func Conds2TableDual(p base.LogicalPlan, conds []expression.Expression) base.LogicalPlan {
```

Comment:
It seems that this function does not belong to logical_join. It shouldn't belong to any logical operator.
It should belong to expression utils.
Could we move it to right place ?

Reply:
I have create new file and move it into new file.